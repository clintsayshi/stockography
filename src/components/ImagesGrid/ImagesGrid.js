import React from "react";
import Card from "../Card/Card";

function ImagesGrid({ images }) {
  return (
    <section
      data-testid="images-grid"
      className="container h-full mx-auto px-4 md:px-8 lg:px-12 xl:px-16"
    >
      <div className="gap-6 grid grid-cols-1 sm:grid-cols-3 lg:grid-cols-4">
        {images.length > 0 &&
          images.map((image, key) => <Card key={key} data={image} />)}
      </div>
    </section>
  );
}

export default ImagesGrid;
