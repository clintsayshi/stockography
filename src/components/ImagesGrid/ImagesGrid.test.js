import renderer from "react-test-renderer";
import { render, screen, cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ImagesGrid from "./ImagesGrid";

afterEach(() => {
  cleanup();
});

describe("Cards of images renderes successfully", () => {
  test("renders cards of images", () => {
    const input = [
      {
        id: 132132,
        tags: "nature, sunrise",
        previewHeight: 250,
        previewWidth: 200,
        previewURL:
          "https://cdn.pixabay.com/photo/2022/04/21/20/49/road-7148369_960_720.jpg",
      },
      {
        id: 132132,
        tags: "street, architecture",
        previewHeight: 250,
        previewWidth: 200,
        previewURL:
          "https://cdn.pixabay.com/photo/2022/04/21/20/49/road-7148369_960_720.jpg",
      },
      {
        id: 132132,
        tags: "nature, flowers",
        previewHeight: 250,
        previewWidth: 200,
        previewURL:
          "https://cdn.pixabay.com/photo/2022/04/21/20/49/road-7148369_960_720.jpg",
      },
    ];

    const component = renderer.create(
      <MemoryRouter>
        <ImagesGrid images={input} />
      </MemoryRouter>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    render(
      <MemoryRouter>
        <ImagesGrid images={input} />
      </MemoryRouter>
    );
    const gridEl = screen.getByTestId("images-grid");
    expect(gridEl).toBeInTheDocument();
  });
});
