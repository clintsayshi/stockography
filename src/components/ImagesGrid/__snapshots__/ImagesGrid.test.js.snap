// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards of images renderes successfully renders cards of images 1`] = `
<section
  className="container h-full mx-auto px-4 md:px-8 lg:px-12 xl:px-16"
  data-testid="images-grid"
>
  <div
    className="gap-6 grid grid-cols-1 sm:grid-cols-3 lg:grid-cols-4"
  >
    <a
      className="relative"
      data-testid="card-132132"
      href="/132132"
      onClick={[Function]}
    >
      <img
        alt="nature, sunrise"
        className="w-full rounded-lg h-60 object-cover"
        height={250}
        src="https://cdn.pixabay.com/photo/2022/04/21/20/49/road-7148369_960_720.jpg"
        width={200}
      />
    </a>
    <a
      className="relative"
      data-testid="card-132132"
      href="/132132"
      onClick={[Function]}
    >
      <img
        alt="street, architecture"
        className="w-full rounded-lg h-60 object-cover"
        height={250}
        src="https://cdn.pixabay.com/photo/2022/04/21/20/49/road-7148369_960_720.jpg"
        width={200}
      />
    </a>
    <a
      className="relative"
      data-testid="card-132132"
      href="/132132"
      onClick={[Function]}
    >
      <img
        alt="nature, flowers"
        className="w-full rounded-lg h-60 object-cover"
        height={250}
        src="https://cdn.pixabay.com/photo/2022/04/21/20/49/road-7148369_960_720.jpg"
        width={200}
      />
    </a>
  </div>
</section>
`;
